includes/Core/Tags/First_Party_Mode/First_Party_Mode.php

```diff
@@ -92,4 +105,36 @@ public function get_debug_fields() {
 			),
 		);
 	}
+
+	/**
+	 * Checks the health of First Party Mode server requirements.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return void
+	 */
+	public function healthcheck() {
+		$is_fpm_healthy           = $this->rest_controller->is_endpoint_healthy( 'https://g-1234.fps.goog/mpath/healthy' );
+		$is_script_access_enabled = $this->rest_controller->is_endpoint_healthy( add_query_arg( 'healthCheck', '1', plugins_url( 'fpm/measurement.php', GOOGLESITEKIT_PLUGIN_MAIN_FILE ) ) );
+
+		$settings    = $this->first_party_mode_settings->get();
+		$fpm_enabled = $settings['isEnabled'];
+
+		$this->first_party_mode_settings->merge(
+			array(
+				'isEnabled'             => $fpm_enabled ? $is_fpm_healthy && $is_script_access_enabled : $fpm_enabled,
```

Comment:
As @hussain-t corrected us [here](https://github.com/google/site-kit-wp/issues/9768#issuecomment-2516233777), we should not be changing the value of `isEnabled` here.

```suggestion
```